FreeRTOS/Test/CMock/tasks/tasks_1_utest.c

```diff
@@ -3594,6 +3617,33 @@ void test_vTask_Set_ThreadLocalStoragePointer_fail( void )
      * */
     TEST_ASSERT_NOT_EQUAL( pValue, value );
 }
+
+void test_vTask_Set_ThreadLocalStoragePointer_failNegative( void )
+{
+    TaskHandle_t task_handle;
+    uint32_t i = 454545;
+    void * pValue = &i;
+
+    /* Setup */
+    create_task_priority = 3;
+    task_handle = create_task();
+    ptcb = task_handle;
+    /* Expectations */
+    /* API Call */
+    vTaskSetThreadLocalStoragePointer( task_handle,
+                                       -1,
+                                       pValue );
+
+    #pragma GCC diagnostic ignored "-Warray-bounds"
+    void * value = *( ptcb->pvThreadLocalStoragePointers - 1 );
```

Comment:
This operation doesn't seem as bad as the one on line 3568, but still seems a bit gross.